clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/model/CloudMetric.groovy

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.model
+
+/**
+ * A marker interface to describe a metric reported by a cloud provider.
+ *
+ * Implementations should add any fields necessary to uniquely identify a particular metric; for example, AWS
+ * supplies "metricName" and "namespace" fields, as well as a collection of "dimensions"
+ */
+interface CloudMetric {
+
```

Comment:
Who consumes this interface? It seems you are creating provider-specific metrics, which means consumers have to know which provider they are using to do anything. That seems to be in conflict (though not entirely) with having the other interfaces for the values.

I'd think a metric would at least have a name, and perhaps a type. What you call the unit below in the CloudMetricStatistics would instead be "metric" and an instance of this. Then if you have some standard types, any consumer can use the metric interface and could render it (having the name) and have a chance at interpreting it (from the type) without having to know the provider. Nonstandard types would still need additional knowledge to interpret (as it is now) but you'd know it is non-standard for whatever that is worth.

I think a type is a unit, but am not sure what units are. Are these things like "bytes" and "seconds" or could they also be rates ("bytes/second"). I think rates are usually introduced higher in the consumption chain.